registry/paraswap/calldata-AugustusSwapper-v5.json

```diff
@@ -0,0 +1,282 @@
+                "label": "Income Amount",
+                "params": {
+                    "tokenPath": "fromToken"
+                }
+            },
+            "exchange": {
+                "format": "addressName",
+                "label": "Exchange address",
+                "params": {
+                    "types": ["wallet"]
+                }
+            }
+        },
+        "formats": {
+            "0x54840d1a" : {
+                "$id": "swapOnUniswap",
+                "fields": [
+                    {
+                        "path": "amountIn",
+                        "$ref": "$.display.definitions.amountIn"
+                    },
+                    {
+                        "path": "amountOutMin",
+                        "$ref": "$.display.definitions.amountOutMin"
+                    },
+                    {
+                        "path": "path.[]",
+                        "label": "Array of addresses",
+                        "format": "addressName",
+                        "params": {
+                            "types": ["wallet"]
+                        }
+                    }
+                ],
+                "required": ["amountIn", "amountOutMin", "path"]
+            },
+            "0xf5661034": {
+                "$id": "swapOnUniswapFork",
+                "fields": [
+                    {
+                        "path": "factory",
+                        "$ref": "$.display.definitions.factory"
+                    },
+                    {
+                        "path": "initCode",
+                        "format": "raw",
+                        "label": "init code"
+                    },
+                    {
+                        "path": "amountIn",
+                        "$ref": "$.display.definitions.amountIn"
+                    },
+                    {
+                        "path": "amountOutMin",
+                        "$ref": "$.display.definitions.amountOutMin"
+                    },
+                    {
+                        "path": "path.[]",
+                        "label": "Array of addresses",
+                        "format": "addressName",
+                        "params": {
+                            "types": ["wallet"]
+                        }
+                    }
+                ],
+                "required": ["factory", "initCode", "amountIn", "amountOutMin", "path"]
+            },
+            "0x935fb84b": {
+                "$id": "buyOnUniswap",
+                "fields": [
+                    {
+                        "path": "amountOut",
+                        "$ref": "$.display.definitions.amountOut"
+                    },
+                    {
+                        "path": "amountInMax",
+                        "$ref": "$.display.definitions.amountInMax"
+                    },
+                    {
+                        "path": "path.[]",
+                        "label": "Array of addresses",
```

Comment:
This does not really describe what are these addresses, is it really what the current plugin is displaying ?

Reply:
This does not seem to be displayed
[https://github.com/LedgerHQ/app-plugin-paraswap/tree/develop/tests/snapshots/ethereum_nanos_buy_on_uniswap](https://github.com/LedgerHQ/app-plugin-paraswap/tree/develop/tests/snapshots/ethereum_nanos_buy_on_uniswap)

Reply:
But it is handled by this function: [https://github.com/LedgerHQ/app-plugin-paraswap/blob/72f602251715bf794184721ed0410f0cd631a5d2/src/handle_provide_parameter.c#L24](https://github.com/LedgerHQ/app-plugin-paraswap/blob/72f602251715bf794184721ed0410f0cd631a5d2/src/handle_provide_parameter.c#L24)